site/content/integrate/apps/quick-start-serverless/_index.md

```diff
@@ -0,0 +1,443 @@
+	http.HandleFunc("/bindings", httputils.HandleJSONData(bindingsData))
+
+	// The form for sending a Hello message.
+	http.HandleFunc("/send/form", httputils.HandleJSONData(formData))
+
+	// The main handler for sending a Hello message.
+	http.HandleFunc("/send/submit", send)
+
+	deployType = dt
+}
+
+func send(w http.ResponseWriter, req *http.Request) {
+	creq := apps.CallRequest{}
+	json.NewDecoder(req.Body).Decode(&creq)
+
+	message := fmt.Sprintf("Hello from a serververless app running as %s!", deployType)
+	v, ok := creq.Values["message"]
+	if ok && v != nil {
+		message += fmt.Sprintf(" ...and %s!", v)
+	}
+
+	asBot := appclient.AsBot(creq.Context)
+	asBot.DM(creq.Context.ActingUserID, message)
+
+	httputils.WriteJSON(w,
+		apps.NewOKResponse(nil, "Created a post in your DM channel."))
+}
+```
+
+#### `go.mod` and `go.sum`
+
+Since package `function` is built independently by OpenFaaS, add its own `go.mod` for its ddependencies:
+
+```go
+module {{path-to-your-module}}/function
+
+go 1.16
+
+require github.com/mattermost/mattermost-plugin-apps v0.8.0
+```
+
+### 4. `./aws`
+#### `main.go`
+```go
+package main
+
+import (
+    "net/http"
+
+    "github.com/aws/aws-lambda-go/lambda"
+    "github.com/awslabs/aws-lambda-go-api-proxy/httpadapter"
+    "github.com/mattermost/mattermost-plugin-apps/apps"
+
+    "{{path-to-your-module}}/function"
+)
+
+func main() {
+    function.InitApp(apps.DeployAWSLambda)
+
+    // run with the Lambda request re-marshaled back into an http.Request so that it works with mux.
+    lambda.Start(httpadapter.New(http.DefaultServeMux).Proxy)
+}
+```
+
+#### `go.mod`
+```go
+module {{path-to-your-module}}/aws
+
+go 1.16
+
+require (
+	github.com/aws/aws-lambda-go v1.19.1
+	github.com/awslabs/aws-lambda-go-api-proxy v0.11.0
+	github.com/mattermost/mattermost-plugin-apps v0.8.0
+	{{path-to-your-module}}/function v0.0.0-00010101000000-000000000000
+)
+
+replace {{path-to-your-module}}/function => ../function
+```
+
+### 5. `./http`
+#### `main.go`
+The HTTP main serves `./static` and `manifest.json` in addition to the app call paths.
+
+```go
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/utils/httputils"
+
+	"{{path-to-your-module}}/function"
+)
+
+func main() {
+	mpath := flag.String("manifest", "", "path to the manifest file to serve on /manifest.json")
+	spath := flag.String("static", "", "path to the static folder to serve on /static")
+	flag.Parse()
+
+	if mpath != nil && *mpath != "" {
+		mdata, err := os.ReadFile(*mpath)
+		if err != nil {
+			panic(err)
+		}
+		http.HandleFunc("/manifest.json", httputils.HandleJSONData(mdata))
+	}
+
+	if spath != nil && *spath != "" {
+		http.Handle("/static/", http.StripPrefix("/static", http.FileServer(http.Dir(*spath))))
+	}
+
+	function.InitApp(apps.DeployHTTP)
+	fmt.Println("Listening on :8080")
+	panic(http.ListenAndServe(":8080", nil))
```

Comment:
Should the code be the same here as for the other examples?

Reply:
0/5 why care? it's valid and idiomatic...